scripts/sbatch.py

```diff
@@ -0,0 +1,476 @@
+    if all2float:
+        value = float(re.findall(r"[-+]?\d*\.\d+|\d+", value)[0])
+    
+    return value
+
+
+def get_value_from_conf(slurm_conf, partition,keyword, all2float=False):
+    """"Read a slurm.conf file and gets the default memory per cpu"""
+
+    with open(slurm_conf, "r") as f:
+        lines = f.readlines()
+    
+    #example line in conf:
+    # PartitionName=DEFAULT PreemptMode=OFF DefMemPerCPU=512 DefaultTime=15 MaxTime=3-00:00 TRESBillingWeights="CPU=0.5,Mem=0.3333G"
+
+    #initializing some values
+    line_with_keyword = []
+    partition_keyword_value = None
+    default_partition_keyword_value = None
+    default_keyword_value = None
+    
+    for line in lines[::-1]:
+        #scan through whole confige file and find lines with keyword
+        if keyword in line:
+            line_with_keyword.append(line)
+    # raise error if none were found
+    if len(line_with_keyword) == 0:
+        raise ValueError("failed to find keyword:{} in config".format(keyword))
+    else:
+        #try find value from partion first. if not, try partition=DEFAULT, else try a global value
+        for line in line_with_keyword:
+            #removes potential unexpected indent
+            line = line.strip()
+            if line.startswith("PartitionName={}".format(partition)):
+                partition_keyword_value = get_value_from_line(line, keyword, all2float) 
+            elif line.startswith("PartitionName={}".format(varLevel.default.id)):
+                default_partition_keyword_value =  get_value_from_line(line, keyword, all2float) 
+            elif line.startswith(keyword):
+                default_keyword_value = get_value_from_line(line, keyword,all2float) 
+        # return a value from narrow to broad
+        if partition_keyword_value is not None:
+            return partition_keyword_value
+        elif default_partition_keyword_value is not None:
+            return default_partition_keyword_value
+        elif default_keyword_value is not None:
+            return default_keyword_value
+        else:
+            #last catch of errors if something unexpected happened
+            raise ValueError("keyword:{} were found but none were parsed correctly".format(keyword))
+
+
+def weight_dict_from_line(line):
+    #replace ',' with ' ' if multiple TRES were defined
+    line = line.replace(",", " ").strip()
+    tmp_dict = []
+    for res in TRES:
+        tmp_dict[res.value] = get_value_from_line(line, res.id, all2float=True)
+    
+    return tmp_dict
+
+
+def billing_weight_from_conf(slurm_conf, partition_name):
+    """
+       Read a slurm.conf and returns the correspoding cpu_weights and mem_weights according to the partition_name
+       if nothing were found, a default of cpu=1.0 mem=0 will be used
+    """
+    with open(slurm_conf, "r") as f:
+        lines = f.readlines()
+    #initialize
+    TRES_dict = {vl.value: None for vl in varLevel}
+    
+    # PartitionName is at the end of file
```

Comment:
Umm, it shouldn't be.

Reply:
this was a logic from melody's.....
I'll fix that.

Reply:
my comment even said scan all lines @.@